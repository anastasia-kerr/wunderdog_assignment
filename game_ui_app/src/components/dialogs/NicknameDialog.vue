<template>
  <v-dialog :width="dialogSize" persistent v-model="dialog">
    <v-form ref="form" v-model="valid">
      <v-card>
        <v-card-title v-text="headerLabel" />
        <v-card-text>
          <v-text-field :label="nicknameLabel" outlined v-model="nickname" :rules="nicknameRules" />
        </v-card-text>
        <v-card-actions>
          <v-spacer />
          <v-btn color="primary" v-text="btnLabel" @click="onBtnPress" />
        </v-card-actions>
      </v-card>
    </v-form>
  </v-dialog>
</template>

<script lang="ts">
import Vue from 'vue'
import { nameIsRequired } from '../../validators'
import config from '../../configs'
const { smallDialogSize } = config

export default Vue.extend({
  name: 'NicknameDialog',

  data: () => ({
    dialog: true,
    valid: false,
    dialogSize: smallDialogSize,
    nickname: '',
    nicknameLabel: 'Nickname',
    nicknameRules: [nameIsRequired],
    headerLabel: 'Choose your display name',
    btnLabel: 'Continue to game'
  }),
  methods: {
    validateForm (): void {
      const form = this.$refs.form as Vue & { validate: any }
      form.validate()
    },
    onBtnPress (): void {
      this.validateForm()
      if (!this.valid) return
      this.dialog = false
      this.$store.dispatch('setUser', this.nickname)
    }
  }
})
</script>
