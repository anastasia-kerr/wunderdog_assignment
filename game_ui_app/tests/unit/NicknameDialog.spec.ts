import Vue from 'vue'
import NicknameDialog from '@/components/dialogs/NicknameDialog.vue'
import { shallowMount, createLocalVue, Wrapper } from '@vue/test-utils'
import Vuetify from 'vuetify'

const localVue = createLocalVue()
Vue.use(Vuetify)

describe('NicknameDialog', () => {
  let vuetify: Vuetify
  let wrapper:Wrapper<Vue>
  beforeEach(() => {
    vuetify = new Vuetify()
    wrapper = shallowMount(NicknameDialog, { localVue, vuetify })
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
