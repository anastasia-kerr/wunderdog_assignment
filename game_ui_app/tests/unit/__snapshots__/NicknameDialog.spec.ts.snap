// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NicknameDialog renders correctly 1`] = `
<v-dialog-stub
  closedelay="0"
  contentclass=""
  opendelay="0"
  openonclick="true"
  origin="center center"
  persistent="true"
  retainfocus="true"
  transition="dialog-transition"
  value="true"
  width="320"
>
  <v-form-stub>
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <v-card-title-stub
        tag="div"
      >
        Choose your display name
      </v-card-title-stub>
       
      <v-card-text-stub
        tag="div"
      >
        <v-text-field-stub
          backgroundcolor=""
          clearicon="$clear"
          errorcount="1"
          errormessages=""
          label="Nickname"
          loaderheight="2"
          messages=""
          outlined="true"
          rules="[Function]"
          successmessages=""
          type="text"
          value=""
        />
      </v-card-text-stub>
       
      <v-card-actions-stub
        tag="div"
      >
        <v-spacer-stub
          tag="div"
        />
         
        <v-btn-stub
          activeclass=""
          color="primary"
          tag="button"
          type="button"
        >
          Continue to game
        </v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-form-stub>
</v-dialog-stub>
`;
